tests/unit/test_repository.py

```diff
@@ -265,6 +267,26 @@ def test_registering_a_new_stack_with_already_registered_components(clean_repo):
     )
 
 
+def test_updating_a_new_stack_with_already_registered_components(clean_repo):
```

Comment:
The test name/description is a little misleading as the new components of this stack are actually not registered beforehand but get registered in the process